python/golem_task_api/testutils/dockertaskapiservice.py

```diff
@@ -1,75 +1,99 @@
+    except Exception:  # pylint: disable=broad-except
         return False
-    return True
+    finally:
+        await docker.close()
 
 
-def run_docker_container(
+async def run_container(
         image: str,
         work_dir: Path,
         command: str,
         port: int
-) -> docker.models.containers.Container:
+) -> DockerContainer:
     """" Start a detached container """
-    if sys.platform == 'linux':
-        ports = {}
-    else:
-        ports = {port: port}
+    ports = {}
+    if sys.platform != 'linux':
+        ports[f'{port}/tcp'] = port
 
-    client = docker.from_env()
-    return client.containers.run(
-        image=image,
-        command=command,
-        volumes={
+    config = dict(
+        Image=image,
+        Cmd=shlex.split(command),
+        Volumes={
             str(work_dir): {
                 'bind': f'/{constants.WORK_DIR}',
                 'mode': 'rw',
             }
         },
-        user=os.getuid(),
-        ports=ports,
-        detach=True,
+        User=str(os.getuid()),
+        HostConfig={
+            'Binds': [f'/{work_dir}:/{constants.WORK_DIR}'],
+            'PortBindings': {
+                f'{port}/tcp': [{
+                    'HostIp': '',
+                    'HostPort': f'{port}'
+                }]
+            },
+        },
+        ExposedPorts={
+            f'{port}/tcp': {}
+        },
+        Detach=True,
     )
 
+    return await Docker().containers.run(config, name=str(uuid.uuid4()))
+
 
-def get_docker_container_port_mapping(
-        container_id: str,
+async def get_docker_container_state(
+        container: DockerContainer,
+) -> str:
+    """ Retrieve container state """
+    container_config = await container.show()
+    return container_config['State']
+
+
+async def get_container_port_mapping(
+        container: DockerContainer,
         port: int,
         vm_name: str,
 ) -> Tuple[str, int]:
     """ Retrieve running container's IP address and port """
 
-    client = docker.APIClient()
-    container_config = client.inspect_container(container_id)
+    container_config = await container.show()
     net_config = container_config['NetworkSettings']
 
     if sys.platform == 'darwin':
         ip_address = '127.0.0.1'
     elif sys.platform == 'win32':
-        ip_address = subprocess.check_output(['docker-machine', 'ip', vm_name])
-        ip_address = ip_address.decode('utf-8').strip()
+        ip_command = ['docker-machine', 'ip', vm_name]
+        ip_output = subprocess.check_output(ip_command)
+        ip_address = ip_output.decode('utf-8').strip()
     else:
         ip_address = net_config['Networks']['bridge']['IPAddress']
 
-    if sys.platform != 'linux':
+    if net_config['Ports']:
         port = int(net_config['Ports'][f'{port}/tcp'][0]['HostPort'])
 
     if not ip_address:
-        raise RuntimeError(f'Unable to read the IP address of {container_id}')
+        raise RuntimeError(f'Unable to read the IP address of {container}')
```

Comment:
will this still print `container_id`?

Reply:
No, this would print an object id. Container id is a protected property of `DockerContainer`, see the new changes on how it's resolved.